### ISSUE TITLE
[WHDLoad] issues

### ISSUE THREAD

**Issue by nemo93** [midwan/amiberry#122]
I can't get the following games to properly run. Most recent WHDLoad packages used for each game. I'm using Retropie 4.3 on a Pi3. If a separate ticket per game is the prefered way please let me know and I'll fill a ticket in for each.

**Read Me first:** 
Please check this [thread](https://retropie.org.uk/forum/topic/12656/amiga-amiberry-whdload-holy-grail-amiberry-s-official-thread/22) on Retropie's forum where you could find **[3 uae templates that should work with almost any games](https://retropie.org.uk/forum/topic/12656/amiga-amiberry-whdload-holy-grail-amiberry-s-official-thread/22)**. First have a look and try to run your games with them before reporting an issue. 
Also ensure **to use the latest version of the WHDload packages** as they are being updated on regular basis with bugs fix and other improvements. You could find the latest packages versions [here on this site](http://www.whdload.de/index.html) (neither roms nor games could be find there). Go to the _**Installs**_ section and pick a letter to find a game version.

-----------------------------------------------------

**Total games listed below: 15
(B) Non-working/unplayable: 6
(G) Working/playable yet w glitches: 9**

-----------------------------------------------------

* **(B) Barbarian The Ultimate Warrior / Palace [version 1.1 (12/03/2017)]** - ECS/OCS
=> game does launch. However impossible to play as it's running at very high speed whatever parameter I set in the uae. Thanks to @maxbeanz from [Retropie's forum](https://retropie.org.uk/forum/topic/12656/amiga-amiberry-whdload-holy-grail-amiberry-s-official-thread/19) for reporting that one.
[uae for Barbarian](https://github.com/midwan/amiberry/files/1293649/BarbarianPalace.txt)

* **(B) Battle Isle [version 1.9 (07.10.17)]** - ECS/OCS
=> game does NOT launch. I got the following message within the whdscript_debug file: `The data/disk-files are damaged or it is an unsupported version.` I've followed all directions given [there](http://www.whdload.de/games/BattleIsle.html) but still no luck. 
**Note:** the version which includes all data disks is not working either and output the exact same error message.
[uae for Battle Isle](https://github.com/midwan/amiberry/files/1356033/Battle.Isle.txt)

* **(B) Body Blows Galactic [version 2.1 (22.05.05)]** - AGA & ECS/OCS
=> game does launch. However impossible to get into a fight as emulator does crash (exit to EmulationStation/Retropie) right after the "vs screen". Just after the characters selection screen.
[uae for Body Blows Galactic](https://github.com/midwan/amiberry/files/1293641/Body.Blows.txt)

* **(G) Cannon Fodder [v1.0 (25.01.2001)]** - ECS/OCS
=> game runs fine. However at the very first intro mission screen where you can see a chopper/helicopter hovering over jungle, you could see that the tail/rotor is not there. This bug can be seen only on this first intro screen, on the following missions tail is there no problem.
[uae for Cannon Fodder](https://github.com/midwan/amiberry/files/1290625/Cannon.Fodder.txt)

* **(G) Dungeon Master 2 [Version 1.0 (03.09.2003)]** - AGA
=> game runs fine. However the loading screen is flickering.
[uae for Dungeon Master 2 AGA](https://github.com/midwan/amiberry/files/1290624/Dungeon.Master.2.txt)

* **(G) New York Warriors [version 1.4 (07.10.17)]** - ECS/OCS
=> game does run absolutely fine however there's no sound effect. To be accurate you can hear sound effects but only once in a while (once every minute or so without any clear patterns of when sounds would be heard).
[uae for New York Warriors](https://github.com/midwan/amiberry/files/1282803/New.York.Warriors.txt)

* **(G) Panza Kick Boxing [version 1.6 (01.07.06)]** - ECS/OCS
=> runs ok yet glitches appear on the boxers characters during an animation/move. Just launch a fight against any opponent, do some moves and the glitches would appear pretty quickly.
[uae for Panza Kick Boxing](https://github.com/midwan/amiberry/files/1282798/Panza.Kick.Boxing.txt)

* **(B) Second Samurai [version 1.1 (01-Feb-2001)]** - AGA
=> graphical mess. Logo and loading screens are fine but as soon as you enter the main menu screen it's messy. It's a bit hard to describe but clearly the colors are not where they are supposed to be! **Please do note that the ECS/OCS version of this game is working like a charm.**
[uae for Second Samurai AGA](https://github.com/midwan/amiberry/files/1282813/Second.Samurai.txt)

* **(G) Shadow Fighter [(Version 1.0 (24.03.2007))]** - AGA
=> runs ok yet too slow (compare with the ECS version to see what the real speed should be). **Please do note that the ECS/OCS version of this game is working like a charm.** Thanks to @vultureg7 and to @ransom1122 for reporting and confirming that one (original ticket #138).
[uae for Shadow Fighter](https://github.com/midwan/amiberry/files/1293641/Body.Blows.txt)

* **(G) Striker [(version 1.0 (01.11.2001))]** - ECS/OCS
=> runs ok yet character sprites flicker during the game. Problem occurs on both WHDLoad & ADF versions of the game. Thanks to @ransom1122 for reporting and confirming that one (original ticket #146).
[uae for Striker](https://github.com/midwan/amiberry/files/1356029/Unreal.txt)

* **(G) Unreal [Version 1.2 (11.03.2008)]** - ECS/OCS
=> runs ok yet glitches appear during the first stage while you're flying the dragon. Each time there's an indication (turn left/right etc) a glitch can be seen instead of the arrow.
[uae for Unreal](https://github.com/midwan/amiberry/files/1356029/Unreal.txt)

* ~~**(B) War Zone (Core Design) [version 2.0 (11-apr-2017)]** - ECS/OCS
=> I do confirm the issue described in #113 by @ransom1122 
[uae for War Zone (Core Design)](https://github.com/midwan/amiberry/files/1282854/War.Zone.Core.Design.txt)~~
**UPDATE**: been fixed in the dev branch (https://github.com/midwan/amiberry/issues/113).

* **(G) Worms Directors Cut [version 1.2 (21.08.2011)]** - AGA
=> UPDATE: game runs fine. However the loading screen is flickering.
[uae for Worms Directors Cut AGA](https://github.com/midwan/amiberry/files/1290623/Worms.Directors.Cut.txt)

* **(G) XP8 [version 1.2 (02.01.03)]** - AGA
=> Text menu on the intro screen is garbled. **Please do note that the ECS/OCS version of this game is working like a charm.**
[uae for XP8 AGA](https://github.com/midwan/amiberry/files/1356086/XP8.txt)

* **(B) Zool [version 2.0 (13-jan-08)]** - AGA
=> graphical mess. Same as Second Samurai. Logo/main screens are fine yet intro and game itself have messed up graphics. Still hard to describe but it's like colors are not properly rendered.
[uae for Zool AGA](https://github.com/midwan/amiberry/files/1282827/Zool.txt)

* **(B) Zool 2 [Version 1.0 (28.05.2002)]** - AGA
=> graphical mess. Same as Second Samurai and Zool. Logo/main screens are fine yet game itself have messed up graphics. Still hard to describe but it's like colors are not properly rendered.
[uae for Zool 2 AGA](https://github.com/midwan/amiberry/files/1282838/Zool.2.txt)

If it's just a matter of setting more appropriate parameters in the config please apologize and I'd be glad if you could share your working uae's here. Much appreciated.

**Comment by Voljega** [comment]
Yep those whdload games used to work fine in UAE4ARM 0.4 but after that something went wrong somewhere between UAE4ARM 0.5 and Amiberry...

I'm sure there is a solution along the way in the future

**Comment by midwan** [comment]
Thanks for the detailed report.
It's fine to keep them in one issue for now, if necessary we'll separate them out later.

I'm in the process of setting up my dev environment again, after a fresh installation. And in the meantime I'm pestering the Microsoft team behind VC++ for Linux to fix some bugs I found there, so we can use their tools to remotely debug (which are free) instead of/in parallel to VisualGDB (which is commercial and not everyone has it). Hopefully that will make it easier of other developers to assist as well. ;)

I know the project has been stalling for a while now, but it's not dead yet.

**Comment by nemo93** [comment]
Thanks a lot @midwan this is much appreciated, really. I'd like to help you with code and stuff yet I'm no dev myself so all I could do for now is to provide clear and detailed reports! I'll keep on testing / toying with some other games and I'll report back the issues (if any) there. At some point, as you mentioned we could split all those issues in different tickets for the sake of clarity.

Good luck, take your time and thanks again for bringing that amazing project to life.

**Comment by blinkydoos** [comment]
All these issues "may" be fixed with TomB's newly released UAE4Arm version. It has features from WinUAE 3.5.0. We'll just have to wait for Midwan/Chips to do a Raspberry Pi port and then test the results. I'm not sure if the new release has Cycle Accuracy ? That may be hoping for too much !! :-)

**Comment by midwan** [comment]
The problem is, TomB's code is not in some source control service. So there's no easy way to find out what changed since the previous versions, you have to do a file diff manually, for each file.

Then decide if that change should be integrated in your code as well, or if it's not relevant (TomB's code is specific for the Pandora platform, so some things don't apply).

It's a time-consuming process... :)

**Comment by blinkydoos** [comment]
@midwan No worries, I was just giving you a heads up that TomB has done a lot of extra work that you may be able to just port (like Lubomyr does with his Android port). That would save you the hassle of writing code that's already been written :) There's no rush.

**Comment by Voljega** [comment]
@midwan provided TomB's code was once not too far from yours could'nt you just check out your 2.1 code then get TomB's code and copy it over your checked out version ?

 That way it would be possible with SourceTree or whatever tool you like to see what's different from your code ?

**Comment by midwan** [comment]
@Voljega I could, but that would show us the differences from his code to mine, which are probably quite extensive. Chips-FR picked up his code, made modifications for the Pi and I took the code from there and made even more modifications.

But you gave me an idea: I could create a repo and drop TomB's old code in, then update it with the newest files and get the changes from there. ;-)

**Comment by blinkydoos** [comment]
@midwan I also notice that a couple of things stopped working in your build compared to Chips UAE4Arm, like this issue https://github.com/midwan/amiberry/issues/64 . It doesn't work in SDL1 or SDL2.

**Comment by midwan** [comment]
@blinkydoos That issue did not occur in my tests on the SDL1 version, it seemed related to the changes in SDL2 only.

When did you notice it stopped working?

**Comment by blinkydoos** [comment]
Yeh, my mistake ! It only happens on SDL2.

**Comment by Voljega** [comment]
@midwan cool ! I didnnt dare to gope you would still have TomB's old code :)

On a side note, CHiPs said on the uae4arm thread on Raspberry's forum that he wasn't interested too much anymore on developing uae4arm for now

**Comment by m693663** [comment]
Hi, I've come across another game that runs a little too fast: - **Baby Jo in Going Home** . Also, the AGA version of **XP8**  has colours all over the place on the Intro Screen and then keeps flickering. The ECS/OCS version works fine. Cheers.

**Comment by nemo93** [comment]
Hey @m693663 thnaks for this feedback. I'll try to reproduce on my local setup. Yet would be so kind to confirm the WHD packages versions you've tested with please? Also if you could share your uae (rename the file with *.txt to upload in this thread) that would be much appreciated. I've already updated the ticket and attached some uae that should be working, just in case. Any more feedback please share!

**Comment by m693663** [comment]
Hi Nemo93. Thank you for your quick reply. Baby Jo is version 1.1 and XP8AGA is version 1.2. I compared scripts, apart from the paths to the files, the realfinegrain_cpu_speed setting in the BabyJo script was set to 1024 in my script. Changing this to 0 as per your script fixed the issue.  As to XP8AGA, I re-installed the files then ran it using your script. The initial screens are fine, once it gets to the XP8 Logo, the screen and sound go off and on in unison. I am able to get to the menu screen with the incorrect colours and start the game, at which point the game runs (and looks good), but the joystick won't work. I ran out of time last night so will test again tonight with your script. To be honest, I'm not worried about XP8, as the ECS version works fine. I am building an arcade machine for my daughter and Baby Jo was one of her favourites way back when. Once again, thank you. You have made a 60 year old amiganut very happy!

**Comment by nemo93** [comment]
Happy to help! re XP8 I'm using a xbox360 controller and it works perfect here.

**Comment by G0nz0uk** [comment]
Hi, I've been testing games too (WHDloads).

Zool 1 and 2 (AGA) - Graphical issues in game as mentioned above
Chaos Engine CD32 - No Intro sound
Batman the Movie - Files missing on bootup 

Will keep checking

**Comment by nemo93** [comment]
Hi @G0nz0uk! Thanks for testing. Batman the Movie is working fine here. Here's [my uae](https://github.com/midwan/amiberry/files/1421335/Batman.-.The.Movie.txt). Please test again and confirm it's working this time. I'll have a look at Chaos Engine CD32 version.

**Comment by nemo93** [comment]
Chaos Engine CD32 no sound confirmed. Also this is expected (unfortunately) as [stated here] (http://www.whdload.de/games/ChaosEngine.html). Based on that it's not related to the emulation itself but to the image (I'm guessing that some musics were in raw formats on the disc and as such can't be put on hdd? keep me honest here).

**Comment by HoraceAndTheSpider** [comment]
Is this CDDA music?

No Whdload supports this because it would need the original CD in the drive to play it. 

 You will need to use an ISO image for Cd music.

**Comment by nemo93** [comment]
hi @HoraceAndTheSpider yes that's my assumption even if I can't confirm that myself.

**Comment by CypherXG** [comment]
No. You don't need an ISO File. ISO is only data. What you need is an cue/Bin File or a ccd (CloneCD) File.
I can confirm that Chaos Enginge has cdda Audio and it works perfect. No issues. 
I'm using ccd imags only.

**Comment by HoraceAndTheSpider** [comment]
Ok, sorry for the use of the wrong terminology. I should have just said “cd  rip” but I think the point was still understood

**Comment by G0nz0uk** [comment]
@nemo93 Batman works great now, thanks!

I like my football games, **Kick Off 2, SWOS 96/97, Emyln Hughes Int Soccer** all work, but: 

**Goal** = has graphical issues in-game which is a shame as this is my favourite, will do some research.

Another game tested:

**IK+** = 99% works but slight graphical issues around the fighters feet.

**Comment by nemo93** [comment]
Awesome news \o/ 
I'll check for Goal but please give a try [with this uae](https://github.com/midwan/amiberry/files/1424222/Sensible.World.Of.Soccer.96-97.txt) from my SWOS and report your findings. Please don't forget to share packages versions as well as your uae's for each non working. That helps :) 

Could you be more specific about IK+ as it's a game I'm playing quite often I haven't seen any glitch so far? Perhaps a screenshot or anything to better understand the issue would be much appreciated! Here's [my uae](https://github.com/midwan/amiberry/files/1424223/International.Karate.txt) for that game. 

Cheers

**Comment by G0nz0uk** [comment]
@nemo93 Let me check the above.  My Goal is Goal_v0.96_1817.zip :(

**Comment by G0nz0uk** [comment]
Apologies if I'm not allowed to post this here (excited), but they have made a Sensible World of Soccer 16-17 version, which is basically the 96-97 version, but with the latest players, on their site is the Amiga (lha) and CD32 versions.  I used my Sensible World of Soccer 96-97 uae file to edit the path names but it won't work. 

I won't post the url in case this cause an issue, but google it and see if you have better luck.
[Sensible World of Soccer 16-17.txt](https://github.com/midwan/amiberry/files/1425528/Sensible.World.of.Soccer.16-17.txt)

**Comment by ransom1122** [comment]
Ill be having a look @ Sensible World of Soccer 16-17 for Amiberry tomorrow..

**Comment by ransom1122** [comment]
Fast Forward time a little and had a quick check.. I have already added this game into my Ultimate Amiga image and it works perfectly. I tried friendly and career mode all good. Attached my config, obviously change your paths for your kicks & games to suit your setup.

[Sensible World of Soccer 16-17 (Earok) (CD32).txt](https://github.com/midwan/amiberry/files/1426504/Sensible.World.of.Soccer.16-17.Earok.CD32.txt)

**Comment by G0nz0uk** [comment]
@ransom1122 - Wow this is great news, I didn't try the CD32 version, seems to be an ISO which see you just pointed to:

cdimage0=/home/pi/RetroPie/roms/amiga-data/Games_CD32/Sensible World of Soccer 16-17 (Earok) (CD32)/CD32Game.iso,image

I didn't know you could do this, I was opening the ISO and copying the contents to a folder and trying to point to that.

**Comment by ransom1122** [comment]
Of course.. Amiberry runs cd32 games. And thats a cd32 image
 All good enjoy. I did hace to tweak the settings a little to avoid ingame graphic glitching but it works 100% now. Enjoy..

**Comment by G0nz0uk** [comment]
@ransom1122 thanks that worked, just get some graphic glitches in some of the menus where is flashes, but game runs fine!  I tried to save my progress and asked for a disk and I didn't know how to create a floppy to save to.  I guess I should just use the save state instead?

**Comment by ransom1122** [comment]
@G0nz0uka The game does not support saving, it is written in the text as it loads at the start. Use savestates I guess.
Menu & Game runs fine at my end..  At times it may glitch here and there, that may be because the game is not an official CD32 release, any homebrew will always be touchy in any Amiga emulator .

**Comment by HoraceAndTheSpider** [comment]
Do they not supply an HDF for this?

There is a non-Whdload (and therefore bug free) SWOS HD installer that has been used for years and would therefore be fine with saving. 

Using save states is no use if you want to create tactics.

**Comment by ransom1122** [comment]
Its one of earoks custom cd32 games. I dont think a hdf exists.

**Comment by HoraceAndTheSpider** [comment]
Pretty sure it comes from here, not Erik.

http://files.swos.eu/swos16-17/

Extracting the LHA and mounting this as an HD will be very easy. I will upload a .uae tomorrow for it. 

I have played a lot of SWOS, including career mode, on Amiberry so I don’t think it needs to take up too much of this thread.

**Comment by HoraceAndTheSpider** [comment]
Be aware there are specific settings which must not be tweeked for the career mode to work properly also.

**Comment by ransom1122** [comment]
Ill have a look at it

**Comment by G0nz0uk** [comment]
@ransom1122 I will use save state, hopefully we can bind a key to that in the future (and load). I only get the glitch on the team list just before the match starts.

What a game though, career mode started (Tottenham - sorry)

**Comment by ransom1122** [comment]
A good friend of mine made a HDF from the LHA of Sens Soccer 16-17 and it works perfect. No glitching. Not sure what I can and can't share on here, but I thought I'd let you know the HDF is the winner!!

**Comment by G0nz0uk** [comment]
@ransom1122 Sorry for my ignorance but what is a HDF?

**Comment by ransom1122** [comment]
@G0nz0uk

Hard Disk File - Single file that Amiberry can boot on a normal Aimga1200 config startup.

Just select a1200 quickstart
Insert HDF in Hard Drives tab
Insert your kickstart
And play

**Comment by Darknior** [comment]
Hi
I have ask @ransom1122 before bugging here but he can't help me.
I have take file on his **Ultimate AMIGA** image to make works WHDLaod games and they crash with black screen after the gray menu :(
I watch this tutorial : https://www.youtube.com/watch?v=dOWqop2Wi1o
And i have all fine and don't understand.
I watch and use this : https://www.youtube.com/watch?v=Kd9BuD28ntU
Now i have some WHDLoad games that warks, but not all like on his collection. And i use his games.
It is really strange, all the path are good ... i don't have any problem with ADF, HDF, CD32 ...
What can i miss ? some help ?
Thanks

**Comment by HoraceAndTheSpider** [comment]
Those tutorials are not designed to use ransoms image or the games he provides. 

The Ultimate Amiga loader for WHDload games is nothing to do with any preloaded images.... even if they did decide to use the same name. Mixing pre-loaded images with other projects is always going to cause problems. 

These projects use the original sources, which are the “RetroPlay” Whdload packs.

I suggest you re-run the auto installer and then follow the online (written) Guide which already has an FAQ that includes this problem. 


Also this is not an Amiberry issue so should not be on this GitHub. There is an entire FaceBook group dedicated to providing support for these projects.

**Comment by HoraceAndTheSpider** [comment]
In many cases , if you load a faulty game a second time, it will tell you what error previously occurred.

Also Consider holding down the “fire” button when booting faulty games in order to force the game to re-scan for the Whdload slave.

**Comment by nemo93** [comment]
Hi @Darknior glad to see you there as well :) Please check this [thread](https://retropie.org.uk/forum/topic/12656/amiga-amiberry-whdload-holy-grail-amiberry-s-official-thread/22) on Retropie's forum where I've shared **[3 uae templates that should work with almost any games](https://retropie.org.uk/forum/topic/12656/amiga-amiberry-whdload-holy-grail-amiberry-s-official-thread/22)**. First have a look and try to run your games with them before reporting an issue. 
Also ensure **to use the latest version of the WHDload packages** as those are being updated regularly fixing bugs and other issues. You could find the latest packages versions [here on this site](http://www.whdload.de/index.html) (neither roms nor games could be find there). Go to the _**Installs**_ section and pick a letter to find the latest package version.

**Comment by Darknior** [comment]
Thanks guys i will try all of that :)
I wish i will repair to make them all working fine ...

Yesterday when i try many games, i don't really see a difference between WHDLoad and HDF games. HDF games are launching faster and (also) have Cheats for some ...
At first i want to use WHDLoad to expect having more cheats for games than on HDF, many are so hard for young children without infinite lives :(

Finally it's same, why using WHDLoad more than HDF ?
I read some thing about game fix ? Not really understand ...

**Comment by nemo93** [comment]
Please let's continue this discussion on [Retropie's forum](https://retropie.org.uk/forum/topic/12656/amiga-amiberry-whdload-holy-grail-amiberry-s-official-thread).

**Comment by blinkydoos** [comment]
It's a shame that all the issues in the first post aren't all seperate issues because some of these are/can be fixed and only nemo93 (or midwan) can edit the post.

Warzone is fixed.
Striker is fixed by changing the blitter.
Barbarian can be fixed by using 68000 and a600 ROM. I posted about it here https://github.com/HoraceAndTheSpider/UAEConfigMaker/issues/25 This may also fix some of the other problems listed in the first post.

**Comment by nemo93** [comment]
Hi @blinkydoos! I'm definitely ready to split up the first post into several tickets (one per game/issue) if that makes everyone life easier. I'd like a confirmation from either @midwan or perhaps @HoraceAndTheSpider that it's the best way for them to track and fix issues. Then I'll go create separate tickets.

@blinkydoos I'm very interested by your uae files for the games listed as non working and by the various parameters you set to get the games working. Let's talk more once separate tickets will be created :)

**Comment by midwan** [comment]
I believe separating them out in different issues is better, so please go ahead if you want :)

**Comment by nemo93** [comment]
Done :) Apologize for the spam!!
